src/deadline/client/ui/widgets/host_requirements_tab.py

```diff
@@ -146,6 +146,42 @@ def get_requirements(self) -> Optional[Dict[str, Any]]:
 
         return requirements
 
+    def set_requirements(self, requirements: Dict[str, Any]):
```

Comment:
I think you've generally got the right approach with all of these values flowing down into subcomponents and being handled there.

Here's what I _think_ we want as our logic for OS requirements. Does this align with your thoughts?
1. If there is no input OS attribute value (e.g. `"attr.worker.os.family"` is not input), clear the field
2. If there is an input OS attribute value (e.g. `"attr.worker.os.family"` is `"linux"`), use that value

Currently, I don't think this code block quite captures all the logic for that. For one, the `os_requirements` returned by `self.os_requirements_box.get_requirements()` would be empty when first creating the host requirements, right? So this might miss those.

I think what we need to do is make constants for the attr names that are special e.g. `"attr.worker.os.family", "attr.worker.cpu.arch"` and filter those out specifically.

Similar note for the hardware requirements.

I think it's doing the right thing for custom requirements, awesome work there :)